lib/delayed/backend/active_record.rb

```diff
@@ -138,23 +138,35 @@ def self.jobs_count(flavor,
         # or opts[:flavor] = <some flavor> to perform on all jobs of that flavor
         def self.bulk_update(action, opts)
           scope = if opts[:flavor]
-            raise("Can't bulk update failed jobs") if opts[:flavor].to_s == 'failed'
-            self.scope_for_flavor(opts[:flavor], opts[:query])
+            raise("Can't #{action.to_s} failed jobs") if opts[:flavor].to_s == 'failed' && action != 'destroy'
+            scopes = []
+            scopes << self.scope_for_flavor(opts[:flavor], opts[:query])
           elsif opts[:ids]
-            self.where(:id => opts[:ids])
+            scopes = []
+            scopes << Delayed::Job.where(:id => opts[:ids])
+            failed_jobs = Delayed::Job::Failed.where(:id => opts[:ids])
+            raise("Can't #{action.to_s} failed jobs") if failed_jobs.any? && action != 'destroy'
```

Comment:
change the `any?` to `exists?` - otherwise it will do a `SELECT COUNT(*)` instead of `SELECT 1 ... LIMIT 1`, the former being more expensive because the db has to iterate every row, instead of simply seeing if there is a row.

also, change the order of the conditional to put `action != 'destroy'` first - that way if it is destroy, it doesn't even bother looking to see if a failed job exists